### PR TITLE
Custom errors pages (with unit test)

### PR DESCRIPTION
**Pull request by Axedez**
**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [ ] I've checked new feature as logged in and logged out user if needed
- [ ] "let!" is used in specs
- [x] screenshot of feature is added if task has UI
Screenshot_1
Screenshot_2
Screenshot_3

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,6 +4,10 @@
 # or operating system, you probably want to add a global ignore instead:
 #   git config --global core.excludesfile '~/.gitignore_global'
 
+# Ignore database.yml
+
+/config/database.yml
+
```
Comment (mmarusyk) [inline .gitignore:10]: Delete this line.

File: app/assets/stylesheets/errors.scss [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      }
+    }
+
+    h3 {
+      font-family: 'Cabin', sans-serif;
+      position: relative;
+      font-size: 16px;
+      font-weight: 700;
+      text-transform: uppercase;
+      color: #262626;
```
Comment (vasya539) [inline app/assets/stylesheets/errors.scss:62]: порадив би виносити всі кольора в /stylesheets/colors.scss, записувати там як змінні, і імпортити

File: app/views/errors/not_found.html.slim [hunk 3]
```diff
@@ -0,0 +1,25 @@
+html[lang="en"]
```
Comment (vasya539) [inline app/views/errors/not_found.html.slim:1]: використовуй лейаут

File: app/views/errors/not_found.html.slim [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  head
+    meta[charset="utf-8"]
+    meta[http-equiv="X-UA-Compatible" content="IE=edge"]
+    meta[name="viewport" content="width=device-width, initial-scale=1"]
+    title
+      | The page you requested was not found
+    link[href="https://fonts.googleapis.com/css?family=Cabin:400,700" rel="stylesheet"]
+    link[href="https://fonts.googleapis.com/css?family=Montserrat:900" rel="stylesheet"]
+    link[type="text/css" rel="stylesheet" href="css/style.css"]
```
Comment (marikrebega) [inline app/views/errors/not_found.html.slim:10]: maybe move header to _header.html.slim and reder it for each errors files

File: app/views/errors/server_error.html.slim [hunk 5]
```diff
@@ -0,0 +1,25 @@
+html[lang="en"]
+  head
+    meta[charset="utf-8"]
+    meta[http-equiv="X-UA-Compatible" content="IE=edge"]
+    meta[name="viewport" content="width=device-width, initial-scale=1"]
+    title
+      | We're sorry, but something went wrong
+    link[href="https://fonts.googleapis.com/css?family=Cabin:400,700" rel="stylesheet"]
+    link[href="https://fonts.googleapis.com/css?family=Montserrat:900" rel="stylesheet"]
+    link[type="text/css" rel="stylesheet" href="css/style.css"]
```
Comment (marikrebega) [inline app/views/errors/server_error.html.slim:10]: the same

File: app/views/errors/unacceptable.html.slim [hunk 6]
```diff
@@ -0,0 +1,25 @@
+html[lang="en"]
+  head
+    meta[charset="utf-8"]
+    meta[http-equiv="X-UA-Compatible" content="IE=edge"]
+    meta[name="viewport" content="width=device-width, initial-scale=1"]
+    title
+      | The change you wanted was rejected
+    link[href="https://fonts.googleapis.com/css?family=Cabin:400,700" rel="stylesheet"]
+    link[href="https://fonts.googleapis.com/css?family=Montserrat:900" rel="stylesheet"]
+    link[type="text/css" rel="stylesheet" href="css/style.css"]
```
Comment (marikrebega) [inline app/views/errors/unacceptable.html.slim:10]: the same

File: spec/controllers/errors_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe ErrorsController, type: :controller do
+  describe '#not_found' do
```
Comment (BogdanBusko) [inline spec/controllers/errors_controller_spec.rb:4]: describe "GET #not_found"

**Review by marikrebega — CHANGES_REQUESTED** [review]

File: app/views/errors/not_found.html.slim (hunk 4 above)
Comment (sviatoslav-krupa) [reply app/views/errors/not_found.html.slim:10]: Don't render `head` in views
`head` must render only in layout

Comment (sviatoslav-krupa) [reply app/views/errors/not_found.html.slim:10]: If you need some fonts on specific pages you can import it in appropriate .scss files

**Review by IvanDanyliv — APPROVED** [review]

File: app/views/errors/server_error.html.slim [hunk 8]
```diff
@@ -0,0 +1,14 @@
+#notfound
```
Comment (BogdanBusko) [inline app/views/errors/server_error.html.slim:1]: why id is notfound?

File: app/views/errors/server_error.html.slim [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+        | Oops! Server Error
+      h1
+        span
+          | 5
+        span
+          | 0
+        span
+          | 0
+    h2
+      | We're sorry, but something went wrong
```
Comment (BogdanBusko) [inline app/views/errors/server_error.html.slim:14]: new line

File: app/views/errors/unacceptable.html.slim [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+        | Oops! Server Error
+      h1
+        span
+          | 4
+        span
+          | 2
+        span
+          | 2
+    h2
+      | The change you wanted was rejected
```
Comment (BogdanBusko) [inline app/views/errors/unacceptable.html.slim:14]: new line

File: app/views/errors/unacceptable.html.slim [hunk 11]
```diff
@@ -0,0 +1,14 @@
+#notfound
```
Comment (BogdanBusko) [inline app/views/errors/unacceptable.html.slim:1]: why id is notfound?

File: spec/controllers/errors_controller_spec.rb [hunk 12]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe ErrorsController, type: :controller do
+  describe 'GET#not_found' do
```
Comment (BogdanBusko) [inline spec/controllers/errors_controller_spec.rb:4]: space after GET

File: spec/controllers/errors_controller_spec.rb [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+    it 'must return status code not found and render new page' do
+      get :not_found
+      expect(response).to have_http_status(:not_found)
+      expect(response).to render_template(:not_found)
+    end
+  end
+
+  describe 'GET#server_error' do
```
Comment (BogdanBusko) [inline spec/controllers/errors_controller_spec.rb:12]: here to

File: spec/controllers/errors_controller_spec.rb [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+    it 'must return status code server error and render new page' do
+      get :server_error
+      expect(response).to have_http_status(:internal_server_error)
+      expect(response).to render_template(:server_error)
+    end
+  end
+
+  describe 'GET#unacceptable' do
```
Comment (BogdanBusko) [inline spec/controllers/errors_controller_spec.rb:20]: and here

File: app/controllers/errors_controller.rb [hunk 15]
```diff
@@ -0,0 +1,13 @@
+class ErrorsController < ApplicationController
+  def not_found
+    render status: :not_found
```
Comment (BogdanBusko) [inline app/controllers/errors_controller.rb:3]: you can write
```ruby
head(:not_found)
```

**Review by Andrii-Kozak — APPROVED** [review]

**Review by Ukhanskyi — APPROVED** [review]

**Review by marikrebega — APPROVED** [review]

**Review by chu4 — APPROVED** [review]

**Pull request merged**